sdk/sdk.go

```diff
@@ -233,10 +228,16 @@ func buildIDPTokenSource(c *config) (auth.AccessTokenSource, error) {
 		c.dpopKey = &rsaKeyPair
 	}
 
+	if c.certExchange != nil && c.tokenExchange != nil {
```

Comment:
Why did you move this condition down?  It seems ideal to fail before potentially generating a new key

Reply:
That was an oversight. Agreed on generating the key pair only when needed.